### PR TITLE
docs: fine-tuning notebook

### LINKED ISSUE #1324: Tutorial on fine-Tuning GPT-3.5-turbo for relevance classifications

### PR DESCRIPTION
**Pull request by axiomofjoy**
resolves #1324

### CODE REVIEW

File: tutorials/evals/evaluate_relevance_fine_tuned_gpt_35_turbo.ipynb [hunk 1]
```diff
@@ -0,0 +1,420 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "prompt_template = PromptTemplate(RAG_RELEVANCY_PROMPT_TEMPLATE_STR)\n",
+    "prompts = fine_tune_df.apply(lambda record: prompt_template.format(record), axis=1).tolist()\n",
```
Comment (RogerHYang) [inline tutorials/evals/evaluate_relevance_fine_tuned_gpt_35_turbo.ipynb:180]:
```suggestion
    "prompts = fine_tune_df.apply(prompt_template.format, axis=1).tolist()\n",
```

File: tutorials/evals/evaluate_relevance_fine_tuned_gpt_35_turbo.ipynb [hunk 2]
```diff
@@ -0,0 +1,420 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "with StringIO() as buffer:\n",
+    "    for example in fine_tune_examples[:50]:  # FIXME\n",
```
Comment (RogerHYang) [inline tutorials/evals/evaluate_relevance_fine_tuned_gpt_35_turbo.ipynb:230]:
```suggestion
    "    for example in fine_tune_examples[:50]:\n",
```

**Review by RogerHYang — APPROVED** [review]

File: tutorials/evals/evaluate_relevance_fine_tuned_gpt_35_turbo.ipynb [hunk 3]
```diff
@@ -0,0 +1,420 @@
...
+    "        buffer.write(json.dumps(example) + \"\\n\")\n",
+    "    buffer.seek(0)\n",
+    "\n",
+    "    file_response = openai.File.create(\n",
```
Comment (RogerHYang) [inline tutorials/evals/evaluate_relevance_fine_tuned_gpt_35_turbo.ipynb:234]: maybe also show how file can be deleted in a comment 
> `# openai.File.delete(sid="file-ArvOj4Yx55jhH1i2tlovlcdY")`

File: tutorials/evals/evaluate_relevance_fine_tuned_gpt_35_turbo.ipynb [hunk 4]
```diff
@@ -0,0 +1,420 @@
...
+   "source": [
+    "fine_tune_df = (\n",
+    "    download_benchmark_dataset(task=\"binary-relevance-classification\", dataset_name=\"wiki_qa-train\")\n",
+    "    .rename(\n",
+    "        columns={\n",
+    "            \"query_text\": \"query\",\n",
+    "            \"document_text\": \"reference\",\n",
+    "        },\n",
+    "    )\n",
+    "    .sample(frac=1.0, random_state=42)\n",
```
[the remark this answers is not in the document]
Comment (axiomofjoy) [reply tutorials/evals/evaluate_relevance_fine_tuned_gpt_35_turbo.ipynb:128]: Probably good to shuffle the fine-tuning data before uploading to the API.

**Pull request closed without merging**